IOPool/Streamer/src/StreamerInputFile.cc

```diff
@@ -179,7 +179,7 @@ namespace edm {
 
   void StreamerInputFile::readStartMessage() {
     using namespace edm::storage;
-    IOSize nWant = sizeof(HeaderView);
+    IOSize nWant = sizeof(InitHeader);
```

Comment:
Correct. I forgot to mention this change in the PR description.
It worked up to now because the rest of the init message is appended to the buffer, and much larger than the difference here, so in practice there were no problems.

It seems, indeed, that the code could read Header (not sure, but maybe I even intended that), but InitHeader is also valid, since it always comes in this case at the beginning of the file.